proxy-robots-module/src/main/java/orwell/proxy/ProxyRobots.java

```diff
@@ -218,7 +240,8 @@ public void run() {
             // We stop the service once there are no more robots
             // connected to the proxy
             while (!Thread.currentThread().isInterrupted() &&
-                    !robotsMap.getConnectedRobots().isEmpty()) {
+                    !robotsMap.getConnectedRobots().isEmpty() &&
```

Comment:
Will do.
As for the sleep inside the thread, I am taking advice from https://docs.oracle.com/javase/tutorial/essential/concurrency/sleep.html

In practice, if I do not put such a sleep, then I make one of my cores 100% busy just for this one thread...